pygments/lexers/mosel.py

```diff
@@ -27,123 +27,400 @@ class MoselLexer(RegexLexer):
             (r'!.*?\n', Comment.Single),
             (r'\(!(.|\n)*?!\)', Comment.Multiline),
             (words((
-                'and', 'array', 'as',
-                'boolean', 'break',
-                'case', 'count', 'counter',
-                'declarations', 'div', 'do', 'dynamic',
-                'elif', 'else', 'end-', 'end', 'evaluation', 'exit',
-                'false', 'forall', 'forward', 'from', 'function',
-                'if', 'imports', 'in', 'include', 'initialisations', 'initializations', 'integer', 'inter', 'is_binary', 'is_continuous', 'is_free', 'is_integer', 'is_partint', 'is_semcont', 'is_semint', 'is_sos1', 'is_sos2',
-                'linctr', 'list',
-                'max', 'min', 'mod', 'model', 'mpvar',
-                'next', 'not', 'of', 'options', 'or',
+                'and', 'as', 'break',
+                'case', 'count',
+                'declarations', 'do', 'dynamic',
+                'elif', 'else', 'end-', 'end', 'evaluation',
+                'false', 'forall', 'forward', 'from', 'function', 'hashmap',
+                'if', 'imports', 'include', 'initialisations', 'initializations', 'inter',
+                'max', 'min', 'model',
+                'namespace', 'next', 'not', 'nsgroup', 'nssearch', 'of', 'options', 'or',
                 'package', 'parameters', 'procedure',
-                'public', 'prod', 'range', 'real', 'record', 'repeat', 'requirements',
-                'set', 'string', 'sum',
+                'public', 'prod', 'record', 'repeat', 'requirements', 'return',
+                'sum',
                 'then', 'to', 'true',
                 'union', 'until', 'uses',
                 'version',
                 'while', 'with'),  prefix=r'\b', suffix=r'\b'), Keyword.Builtin),
-            (r'(integer|is_binary|real)', Keyword.Type),
-            (r'(\+|\-|\*|/|=|<=|>=|==|\||\^|<|>|<>|\.\.|:=|:)', Operator),
+            (words((
+                'range', 'array', 'set', 'list', 'mpvar', 'mpproblem', 'linctr', 'nlctr', 'integer',
+                'string', 'real', 'boolean', 'text', 'time', 'date', 'datetime', 'returned', 'Model',
+                'Mosel', 'counter', 'xmldoc',
+                'is_sos1', 'is_sos2', 'is_integer', 'is_binary', 'is_continuous',
+                'is_free', 'is_semcont', 'is_semint', 'is_partint'
+                )), Keyword.Type),
+            (r'(\+|\-|\*|/|=|<=|>=|==|\||\^|<|>|<>|\.\.|\.|:=|:|::|in|mod|div)', Operator),
```

Comment:
`:` and `::` need to swap places to have `::` recognized.

Reply:
Thanks for the hint. I just reordered the `:` and `::`.